src/Nethermind/Nethermind.Network.Discovery/Kademlia/Kademlia.cs

```diff
@@ -36,6 +37,8 @@ public class Kademlia<TNode, TContentKey, TContent> : IKademlia<TNode, TContentK
     private readonly TimeSpan _refreshInterval;
     private readonly ILogger _logger;
 
+    private bool _useTreeImplementation = true;
```

Comment:
No need for a flag. Switch to it directly.

Reply:
Done.

Reply:
You know what? On second thought, this might be a good idea.